client/src/main/java/tds/assessment/Assessment.java

```diff
@@ -12,9 +12,18 @@
     private String selectionAlgorithm;
     private float startAbility;
     private String subject;
+    private List<ItemConstraint> itemConstraints;
```

Comment:
I agree with initializing all these as empty lists (to avoid NPEs). Shouldn't in happen in practice since the query() repo calls return an empty list by default, but still a good idea.

I'm not really convinced that arrays are necessary here. I guess they wouldn't hurt, but from my experience/research they don't really seem to have much benefit. When working with objects I prefer to use Lists. Lists have more flexibility, and I don't think we'd see any sort of performance gain by switching to arrays. My understanding is that the ArrayList implementation uses arrays internally anyway. With that said, I'm open and could definitely be convinced to use arrays instead.